### ISSUE TITLE
ERROR 'Sink socket (7)  got error: 11' ...

### ISSUE THREAD

**Issue by PreinfalkG** [alexander-sholohov/rtlmuxer#1]
Hi

First, thank you for your work!

I have tried the RTLMUXER in this Use Case 
 > [https://docs.google.com/drawings/d/12y-5o1NRYhRyUOC0G6iU-PfSuIVwE-fQxaioNZ806mI/edit?usp=sharing ](https://docs.google.com/drawings/d/12y-5o1NRYhRyUOC0G6iU-PfSuIVwE-fQxaioNZ806mI/edit?usp=sharing)

Basically the whole thing works.
But I get a lot of these error messages: **Sink socket (7)  got error: 11** 
see screenshot under: [https://drive.google.com/open?id=1bv4sUin9_Hm_YcgkeJurR9B_18rpq4hf](https://drive.google.com/open?id=1bv4sUin9_Hm_YcgkeJurR9B_18rpq4hf) 

### **who has an idea for it?**

My Device is a Raspberry Pi 3
- PRETTY_NAME=Raspbian GNU/Linux 9 (stretch)
- NAME= Raspbian GNU/Linux
- VERSION_ID=9
- VERSION=9 (stretch)
- ID=raspbian
- ID_LIKE=debian
- HOME_URL=http://www.raspbian.org/

Thanks
GuenterP

**Comment by alexander-sholohov** [comment]
Hello! Thank you for detail environment description!
The error 11(EAGAIN) occurs when the socket is still busy sending the previous data. This is not a big problem if you see this error from time to time (let's say once per second), otherwise this means lack of CPU power to handle all tasks.
Is it posible to decrease sample rate? For example down to 1.024Msps?
Do you hear any clicks in sound from SDR#? 
Which process consumes most of CPU?
Thank you.

**Comment by PreinfalkG** [comment]
Hi

Thank you for your response!

The error unfortunately occurs continuously.
I am not a Linux expert, but i believe that the processor load is not yet at its limits

So that you can see this ‘LIVE’, I have created a short screen video under [https://youtu.be/6gsiZVPGnmo](https://youtu.be/6gsiZVPGnmo)

You only need the first three terminals windows on the left side.
1.	RTL_TCP process
2.	Your RTLMUXER
3.	My client (SONDEUDP-0) that connects to the Sink port A


Until second 6-7 in the video only the client 'SONDEUDP-0' is connected to the Skink port A.
This client on 'Sink Port A' configures/controls the RTL_TCP process. Already here the error occurs for the first time (until second 6 in the video this one error message is visible)

From second 7 I connect with SDRSharp to the Skink Port B (remotely from my pc). From this point on the errors occur continuously as long as I am connected with SDRSharp.

Also the processor load should be visible in the video ...

Note: decrease sample rate to 1.024Msps is not possible in my case because the program at the 'Shik Port A' needs the sample rate of 2.048 to decode weather ballons.

Thank your for your time

**Comment by alexander-sholohov** [comment]
Hello,
Indeed, the CPU itself is not overloaded, possible the network subsystem is.
I've made small changes in the code, at now it will print just char 'o' instead of long error. Actually this not an error, just a signal that the system is about to overload.
I looked through dxlAPRS code. Possible, we can change the sample rate. File radio.c line 335 [link](https://github.com/oe5hpm/dxlAPRS/blob/e6f0309724c9ad7510368b81d24f75ecc32ee367/src/radio.c#L335) . Try to replace 2048000UL to 1024000UL
